rdkit/Chem/Draw/SimilarityMaps.py

```diff
@@ -98,18 +97,14 @@ def GetAtomicWeightsForModel(probeMol, fpFunction, predictionFunction):
       fpFunction -- the fingerprint function
       predictionFunction -- the prediction function of the ML model
     """
-  if hasattr(probeMol, '_fpInfo'):
-    delattr(probeMol, '_fpInfo')
-  probeFP = fpFunction(probeMol, -1)
-  baseProba = predictionFunction(probeFP)
+
+  _CleanFpInfoAttr_(probeMol)
+  baseProba = predictionFunction(fpFunction(probeMol, -1))
+
   # loop over atoms
-  weights = []
-  for atomId in range(probeMol.GetNumAtoms()):
-    newFP = fpFunction(probeMol, atomId)
-    newProba = predictionFunction(newFP)
-    weights.append(baseProba - newProba)
-  if hasattr(probeMol, '_fpInfo'):
-    delattr(probeMol, '_fpInfo')
+  weights = [baseProba - predictionFunction(fpFunction(probeMol, atomId)) for atomId in range(probeMol.GetNumAtoms())]
```

Comment:
I prefer writing long list comprehensions like this over two lines
```
weights = [baseProba - predictionFunction(fpFunction(probeMol, atomId)) 
           for atomId in range(probeMol.GetNumAtoms())]
```

Reply:
Agree ! I would fix it in the next patch